.github/workflows/maven.yaml

```diff
@@ -96,148 +114,191 @@ jobs:
       - name: checkout project
         uses: actions/checkout@v2
 
-      - name: setup project
-        uses: ./.github/workflows/composites/setup
-
       - name: set env variables
         uses: ./.github/workflows/composites/env-variables
 
+      - name: setup project jdk-17
+        uses: ./.github/workflows/composites/setup-jdk17
+        if: env.BASE_BRANCH == 'main'
+
+      - name: setup project jdk-8
+        uses: ./.github/workflows/composites/setup-jdk1.8
+        if: env.BASE_BRANCH == '2.1.x'
+
       - name: cache local maven repository
         uses: ./.github/workflows/composites/cache
 
       - name: download docker images
         uses: ./.github/workflows/composites/download-docker-images
+        if: env.BASE_BRANCH != '2.1.x'
 
       - name: load docker images into local repo
         uses: ./.github/workflows/composites/load-docker-images
+        if: env.BASE_BRANCH != '2.1.x'
 
       - name: run tests
         env:
           CURRENT_INDEX: ${{ matrix.current_index }}
           NUMBER_OF_JOBS: ${{ matrix.number_of_jobs }}
         run: |
-
-            # - find all tests
-            # - exclude Fabric8IstionIT
-            # - only take classes that have @Test inside them
-            # - ignore the ones that have 'abstract class'. we do this because otherwise we would pass
-            #   to -DtestsToRun an abstract class, and it will not run anything.
-            # - drop the "begining" xxx/src/test/java
-            # - replace / with .
-            # - drop last ".java"
-            # - replace newline with space
-            # - replace '\n' with ' '
-
+          
```

Comment:
there are two changes only in this script:

- exclude all integration tests for `2.1.x` branch (this comment : `# exclude all integration tests for branch '2.1.x' via 'grep -v '.*IT.java''`)
- exclude some projects when building `2.1.x` and running tests, via:

```
if [[ $baseBranch == "2.1.x" ]]; then
          
            ./mvnw -s .settings.xml -pl '-:kubernetes-leader-election-example' -pl '-:kubernetes-hello-world-example' \
            -pl '-:kubernetes-reload-example' -pl '-:kubernetes-loadbalancer-example' \
            -pl '-:spring-cloud-kubernetes-configserver' -pl '-:spring-cloud-kubernetes-configuration-watcher' \
            -pl '-:spring-cloud-kubernetes-discoveryserver' \
```